src/module.c

```diff
@@ -229,7 +227,13 @@ int TSDB_info(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 }
 
 void _TSDB_queryindex_impl(RedisModuleCtx *ctx, QueryPredicateList *queries) {
-    RedisModuleDict *result = QueryIndex(ctx, queries->list, queries->count);
+    bool hasPermissionError = false;
+    RedisModuleDict *result = QueryIndex(ctx, queries->list, queries->count, &hasPermissionError);
```

Comment:
Yes, it will be deleted automatically in this case, as the automemory is on. For the consistency with all other places, I don't do this manually here as well.